src/components/Notifications/RuleSimple/RuleSimple.js

```diff
@@ -223,12 +216,129 @@ export default class RuleSimple extends React.Component {
       this.state.secondInputValue?.length
     ) {
       this.validateSecondQuery()
+      this.handleResetSecondQueryJoinColumns()
     }
     if (this.state.secondQueryResponse && this.state.secondQueryResponse !== prevState.secondQueryResponse) {
       setTimeout(() => {
         this.secondFieldSelectionGridRef?.current?.scrollIntoView({ behavior: 'smooth', block: 'end' })
       }, 100)
     }
+    if (this.state.secondQueryResult !== prevState.secondQueryResult && this.state.secondTermType === QUERY_TERM_TYPE) {
+      this.updateSecondFieldSelectionGrid()
+    }
+    if (this.state.secondQueryResult !== prevState.secondQueryResult && this.state.storedInitialData !== undefined) {
+      this.setState((prevState) => ({
```

Comment:
Could this logic be moved to a function to increase readability? A function like handleResetSecondQueryJoinColumns